ballerina/modules/wssec/encryption.bal

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/crypto;
+import ballerina/jballerina.java;
+
+isolated class Encryption {
+
+    private handle nativeEncryption;
+
+    isolated function init() returns Error? {
+        self.nativeEncryption = newEncryption();
+    }
+
+    isolated function encryptData(string dataString, EncryptionAlgorithm encryptionAlgorithm,
+                         crypto:PublicKey|crypto:PrivateKey key) returns byte[]|Error {
+        byte[] data = dataString.toBytes();
+        do {
+            return check crypto:encryptRsaEcb(data, key);
+        } on fail var e {
+            return error(e.message());
+        }
```

Comment:
Isn't the on fail block redundant here? can't we directly `return crypto:encryptRsaEcb(data, key);`

Reply:
check similar places

Reply:
Fixed in [902844f](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/902844feeea46ae121bac4d6f4127613ef2cc750)